src/wui/wui_api.c

```diff
@@ -29,54 +29,69 @@ void get_telemetry_data(char *data, const uint32_t buf_len) {
     }
     osMutexRelease(wui_thread_mutex_id);
 
-    int32_t actual_nozzle = (int32_t)(wui_vars_copy.temp_nozzle);
-    int32_t actual_heatbed = (int32_t)(wui_vars_copy.temp_bed);
-    double z_pos_mm = (double)wui_vars_copy.pos[Z_AXIS_POS];
-    uint16_t print_speed = (uint16_t)(wui_vars_copy.print_speed);
-    uint16_t flow_factor = (uint16_t)(wui_vars_copy.flow_factor);
+    float z_pos_mm = wui_vars_copy.pos[Z_AXIS_POS];
     const char *filament_material = filaments[get_filament()].name;
 
     if (!wui_vars_copy.sd_printing) {
         snprintf(data, buf_len, "{"
-                                "\"temp_nozzle\":%ld,"
-                                "\"temp_bed\":%ld,"
+                                "\"temp_nozzle\":%.2f,"
```

Comment:
This function is also used for /api/telemetry response (local page does not support these values), so we should probably make separate functions for connect and for local page or inform Bruno about the updates